rmw_fastrtps_cpp/src/qos.cpp

```diff
@@ -157,4 +157,32 @@ get_datawriter_qos(
 
   return true;
 }
+
+bool
+is_time_default(
+  const rmw_time_t & time)
+{
+  return time.sec == 0 && time.nsec == 0;
+}
+
+bool
+is_valid_qos(
+  const rmw_qos_profile_t & qos_policies)
+{
+  if (!is_time_default(qos_policies.deadline)) {
```

Comment:
There should be an enum for deadline and lifespan that you could check as well. They should be set to NONE by default. Long term it probably makes sense to consider that different from a zero initialized time (which may still be considered invalid). 

I wouldn't consider this a blocker by any means.

Reply:
Hmm, does that mean we want to add one? As these do not currently exist in the RMW layer

Reply:
We discussed this offline and we're not going to add the enums to the rmw package for deadline and lifespan at this point.